great_expectations/data_context/store/ge_cloud_store_backend.py

```diff
@@ -522,10 +522,16 @@ def remove_key(self, key):
             )
 
     def _has_key(self, key: Tuple[str, ...]) -> bool:  # type: ignore[override]
-        # self.list_keys() generates a list of length 2 tuples
-        if len(key) == 3:
-            key = key[:2]
-        all_keys = self.list_keys()
+        # Due to list_keys being inconsistently sized (due to the possible of resource names),
+        # we remove any resource names and assert against key ids.
+
+        def _shorten_key(key) -> Tuple[str, str]:
+            if len(key) > 2:
+                key = key[:2]
+            return key
+
+        key = _shorten_key(key)
+        all_keys = set(map(lambda key: _shorten_key(key), self.list_keys()))
```

Comment:
What's behind the lambda? Can you just map `_shorten_key` directly?

Reply:
ah very true haha I originally used a lambda to parse out `key[1]` but you are right!